src/smt/env.cpp

```diff
@@ -282,4 +289,20 @@ theory::quantifiers::OracleChecker* Env::getOracleChecker() const
   return d_ochecker.get();
 }
 
+void Env::registerBooleanTermSkolem(const Node& k)
+{
+  Assert(k.isVar());
+  d_boolTermSkolems.insert(k);
+}
+
+bool Env::isBooleanTermSkolem(const Node& k) const
+{
+  // optimization: check whether k is a variable
+  if (!k.isVar())
+  {
+    return false;
+  }
```

Comment:
why can't variables be boolean term skolems?

Reply:
There is a `!`, i.e. only variables are Boolean term skolems.